scripts/src/javatests/org/oppia/android/scripts/common/MavenDependenciesListGeneratorTest.kt

```diff
@@ -0,0 +1,1330 @@
+  private val LICENSE_DETAILS_INCOMPLETE_FAILURE = "Licenses details are not completed"
+  private val UNAVAILABLE_OR_INVALID_LICENSE_LINKS_FAILURE =
+    "License links are invalid or not available for some dependencies"
+  private val SCRIPT_PASSED_MESSAGE =
+    "Script executed succesfully: maven_dependencies.textproto updated successfully."
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  private val mockLicenseFetcher by lazy { initializeLicenseFetcher() }
+  private val commandExecutor by lazy { initializeCommandExecutorWithLongProcessWaitTime() }
+  private val mavenDependenciesListGenerator = MavenDependenciesListGenerator(
+    mockLicenseFetcher,
+    commandExecutor
+  )
+  private lateinit var testBazelWorkspace: TestBazelWorkspace
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("scripts", "assets")
+    tempFolder.newFolder("third_party")
+    testBazelWorkspace = TestBazelWorkspace(tempFolder)
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testRetrieveThirdPartyMavenDepsList_oneDepInDepGraph_returnsCorrectDep() {
+    val coordsList = listOf(DEP_WITH_SCRAPABLE_LICENSE)
+    testBazelWorkspace.setUpWorkspaceForRulesJvmExternal(coordsList)
+    val thirdPartyPrefixCoordList = coordsList.map { coordinate ->
+      "//third_party:${omitVersionAndReplaceColonsPeriods(coordinate)}"
+    }
+    createThirdPartyAndroidBinary(thirdPartyPrefixCoordList)
+    writeThirdPartyBuildFile(coordsList)
+    val depsList = mavenDependenciesListGenerator.retrieveThirdPartyMavenDependenciesList(
+      "${tempFolder.root}"
+    )
+    assertThat(depsList).contains(
+      omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_SCRAPABLE_LICENSE)
+    )
+  }
+
+  @Test
+  fun testRetrieveThirdPartyMavenDepsList_multipleDepsInDepGraph_returnsCorrectDeps() {
+    val coordsList = listOf(
+      DEP_WITH_SCRAPABLE_LICENSE,
+      DEP_WITH_INVALID_LINKS,
+      DEP_WITH_DIRECT_LINK_ONLY_LICENSE
+    )
+    testBazelWorkspace.setUpWorkspaceForRulesJvmExternal(coordsList)
+    val thirdPartyPrefixCoordList = coordsList.map { coordinate ->
+      "//third_party:${omitVersionAndReplaceColonsPeriods(coordinate)}"
+    }
+    createThirdPartyAndroidBinary(thirdPartyPrefixCoordList)
+    writeThirdPartyBuildFile(coordsList)
+    val depsList = mavenDependenciesListGenerator.retrieveThirdPartyMavenDependenciesList(
+      "${tempFolder.root}"
+    )
+    assertThat(depsList).contains(
+      omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_SCRAPABLE_LICENSE)
+    )
+    assertThat(depsList).contains(
+      omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_INVALID_LINKS)
+    )
+    assertThat(depsList).contains(
+      omitVersionAndReplaceColonsHyphensPeriods(
+        DEP_WITH_DIRECT_LINK_ONLY_LICENSE
+      )
+    )
```

Comment:
```suggestion
    assertThat(depsList).contains(
      omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_SCRAPABLE_LICENSE),
      omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_INVALID_LINKS),
      omitVersionAndReplaceColonsHyphensPeriods(
        DEP_WITH_DIRECT_LINK_ONLY_LICENSE
      )
    )
```

Bit more idiomatic. Ditto elsewhere.

(You can also test order by adding ``.inOrder()`` after the ``contains()`` call).

Reply:
This isn't working - 
![Screenshot from 2021-08-05 15-19-06](https://user-images.githubusercontent.com/54636525/128347858-d6493d13-ab8d-4bae-8892-41b9e4824f01.png)

Reply:
No longer relevant per latest changes.